README.md

```diff
@@ -131,15 +141,364 @@ class Admin < Dill::Role
+  end
+
+  # with the form macro
+  form :todo_form, '.todo-form' do
+  end
+```
+
+
+## Form Fields
+Forms can have form-field widgets defined, as well as any regular sub-widgets. Form field widgets by default try to match their second argument with the text of a label, an input name or an input id.
+
+```ruby
+  form :form_with_everything, '.form-with-everything' do
+    # text field
+    text_field :request, 'request'
+
+    # select field
+    select :state, 'state'
+
+    # checkbox field
+    check_box :receive_email, 'receive_email'
+
+    # regular sub-widget
+    widget :hidden_field, '.hidden-field'
+
+    # currently no built-in form support for radio buttons.
+  end
+```
+
+
+#### Defining Fields with CSS Selectors
+If you'd rather use a CSS selector, you can do that by passing the second argument as an array, with the selector as the array's first element.
+
+```ruby
+  text_field :request, ['.request-field-class'],
+```
+
+
+#### Getting Field Values
+You can get the current value of any of these elements at any time by referencing them on the form object.
+
+```
+  widget(:form_with_everything).state #=> 'CO'
+```
+
+
+## Submitting a Form
+Dill will easily submit a form for you, via the UI, with either of these methods.
+
+```ruby
+  params = { request: 'Call me', state: 'CO', receive_email: true }
+
+  # the submit macro
+  submit :form_with_everything, params
+
+  # the submit widget method
+  widget(:form_with_everything).submit_with(params)
+```
+
+You can also submit a form by setting the fields individually.
+```ruby
+  widget(:form_with_everything).tap do |f|
+    f.request = 'Call me'
+    f.state = 'CO'
+    f.receive_email = true
+    f.submit
+  end
+```
+
+
+#Field Groups<a name="field_groups"></a>
+A field group is like a form without a submit button, and is more semantically appropriate for forms that are not submitted traditionally. It has all the same functionality as a Form object, minus the submitting function.
+
+
+## Field Group Declaration
+```ruby
+  widget :todo_form, '.todo-form', Dill::FieldGroup do
+  end
+```
+
+
+# Lists<a name="lists"></a>
+A list is a collection of Dill ListItem objects, and is a good way to map a simple list format with a Dill element. (For more complicated lists, consider the [table](#tables) element.)
+
+
+## List Declaration
+```ruby
+  # with explicit class
+  class TodoList < Dill::List
+    root '.todo-list'
+  end
+
+  # with widget macro and explicit class
+  widget :todo_list, '.todo-list', Dill::List do
+  end
+
+  # with the list macro
+  list :todo_list, '.todo-list' do
+  end
+```
+
+
+## List Items
+A List has access to its items, defined with a CSS selector.
+
+```ruby
+  list :todo_list, '.todo-list' do
+    item 'li'
+  end
+```
+
+List items are just a form of widget, and can have methods defined on them.
+
+```ruby
+  list :todo_list, '.todo-list' do
+    item 'li' do
+      def downcased
+        text.downcase
+      end
+    end
+  end
+```
```

Comment:
It may also be worth pointing out that, when you only care about the items, but not the list itself, you can do something like 

```ruby
widget :todo_list_item, ".todo-list li"
```

Of course, if you try to use `widget(:todo_list_item)` here, when there is more than one list item, you'll get an error. but you can use `widgets(:todo_list_item)` which will return a list of all items. Keep in mind that this doesn't wait like `widget(:todo_list).items`, though, which at least expects `:todo_list` to be present.